### ISSUE TITLE
Limited number of tests collected/discovered?

### ISSUE THREAD

**Issue by egabrum** [DonJayamanne/pythonVSCode#927]
## Environment data
VS Code version:  1.11.2
Python Extension version: 0.6.3  
Python Version:   2.7.13 32-bit
OS and version:    Windows 7

## Actual behavior
I have a PyTest project with 4527 tests (many parametrized tests) and it's going to get much, much larger.
VS Code only seems to collect about 3000 of them. 
If I go: F1 > Python: Debug Unit Test Method... (I do this because the test lenses don't show up on my tests), I can only find tests listed in the  "Python Test Log" output terminal which, seems to be a incomplete collection list with only about 3000 of my tests. 

## Expected  behavior
I expect to see the Run Test | Debug Test lens on my tests... but that is a different problem.
What I really expect is for VS Code to collect ALL my test in the project and allow me to debug them.

## Steps to reproduce:  
-  Create a project with 4000+ pytest tests
-  Check the  "Python Test Log" output terminal to verify that not all tests are collected.

## Settings  
Your launch.json (if dealing with debugger issues):  
```json
     {
         "name": "Python",
         "type": "python",
         "request": "launch",
         "stopOnEntry": true,
         "pythonPath": "${config.python.pythonPath}",
         "program": "${file}",
         "cwd": "${workspaceRoot}",
         "env": null,
         "envFile": "${workspaceRoot}/.env",
         "debugOptions": [
            "WaitOnAbnormalExit",
            "WaitOnNormalExit",
            "RedirectOutput"
         ]
      },
      {
         "name": "PySpark",
         "type": "python",
         "request": "launch",
         "stopOnEntry": true,
         "osx": {
            "pythonPath": "${env.SPARK_HOME}/bin/spark-submit"
         },
         "windows": {
            "pythonPath": "${env.SPARK_HOME}/bin/spark-submit.cmd"
         },
         "linux": {
            "pythonPath": "${env.SPARK_HOME}/bin/spark-submit"
         },
         "program": "${file}",
         "cwd": "${workspaceRoot}",
         "env": null,
         "envFile": "${workspaceRoot}/.env",
         "debugOptions": [
            "WaitOnAbnormalExit",
            "WaitOnNormalExit",
            "RedirectOutput"
         ]
      },
      {
         "name": "Python Module",
         "type": "python",
         "request": "launch",
         "stopOnEntry": true,
         "pythonPath": "${config.python.pythonPath}",
         "module": "module.name",
         "cwd": "${workspaceRoot}",
         "env": null,
         "envFile": "${workspaceRoot}/.env",
         "debugOptions": [
            "WaitOnAbnormalExit",
            "WaitOnNormalExit",
            "RedirectOutput"
         ]
      },
```   
Your settings.json:   
```json
"python.pythonPath": "c:/Python27/python.exe",
   "python.unitTest.nosetestsEnabled": false,
   "python.unitTest.promptToConfigure": true,
   "python.unitTest.pyTestArgs": ["-vs", "--color=yes"],
   "python.unitTest.pyTestEnabled": true,
   "python.unitTest.unittestEnabled": false,
```   
## Logs
Output from ```Python``` output panel  
```
```   
Output from ```Console window``` (Help->Developer Tools menu)
```
keybindingsRegistry.ts:195Ctrl+Alt+ keybindings should not be used by default under Windows. Offender:  ealtKey: truectrlKey: truekeyCode: 41metaKey: falseshiftKey: falsetype: 1__proto__: Object  for  bookmarks.toggle
keybindingsRegistry.ts:195Ctrl+Alt+ keybindings should not be used by default under Windows. Offender:  e  for  bookmarks.jumpToNext
keybindingsRegistry.ts:195Ctrl+Alt+ keybindings should not be used by default under Windows. Offender:  e  for  bookmarks.jumpToPrevious
keybindingsRegistry.ts:195Ctrl+Alt+ keybindings should not be used by default under Windows. Offender:  e  for  extension.toggleRegexPreview
keybindingsRegistry.ts:195Ctrl+Alt+ keybindings should not be used by default under Windows. Offender:  e  for  revisionmarks.toggle
keybindingsRegistry.ts:195Ctrl+Alt+ keybindings should not be used by default under Windows. Offender:  e  for  revisionmarks.jumpToNext
keybindingsRegistry.ts:195Ctrl+Alt+ keybindings should not be used by default under Windows. Offender:  e  for  revisionmarks.jumpToPrevious
keybindingResolver.ts:129Conflict detected, command `editor.action.goToImplementation` cannot be triggered due to C_Cpp.GoToDeclaration
keybindingResolver.ts:129Conflict detected, command `workbench.action.terminal.openNativeConsole` cannot be triggered due to startanyshell.startShell
keybindingResolver.ts:129Conflict detected, command `editor.action.insertLineAfter` cannot be triggered due to task.new
keybindingResolver.ts:129Conflict detected, command `task.complete` cannot be triggered due to tasks.completeTask
keybindingResolver.ts:129Conflict detected, command `task.cancel` cannot be triggered due to tasks.convertTask
extensionHost.ts:284[Extension Host] [Guides] Sending usage statistics...
extensionHost.ts:284[Extension Host] Revisionmarks is activated
extensionHost.ts:284[Extension Host] [Guides] Error while sending usage statistics: Error: getaddrinfo ENOTFOUND api.digitalparticle.com api.digitalparticle.com:80
extensionHost.ts:284[Extension Host] [Guides] Usage statistics will retry in the next 5 minutes
2shell.ts:475Cannot read property 'revisionmarks' of undefined: TypeError: Cannot read property 'revisionmarks' of undefined
    at C:\Users\essa78\.vscode\extensions\yuhuagou.Revisionmarks-0.1.0\extension.js:77:23
    at e.invoke (c:\Program Files (x86)\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:4:16423)
    at e.fire (c:\Program Files (x86)\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:4:20054)
    at c:\Program Files (x86)\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:4:305758
    at e.invoke (c:\Program Files (x86)\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:4:16423)
    at e.fire (c:\Program Files (x86)\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:4:20054)
    at t.$acceptDocumentsAndEditorsDelta (c:\Program Files (x86)\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:4:310824)
    at t.e.handle (c:\Program Files (x86)\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:4:277319)
    at s (c:\Program Files (x86)\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:4:135077)
    at h (c:\Program Files (x86)\Microsoft VS Code\resources\app\out\vs\workbench\node\extensionHostProcess.js:4:135754)e.onUnexpectedError @ shell.ts:475
extensionHost.ts:284[Extension Host] ApplicationInsights:Sender [Object]0: Objectlength: 1__proto__: Array[0]
extensionHost.ts:284[Extension Host] ApplicationInsights:Sender [Object]
extensionHost.ts:284 [Extension Host] ApplicationInsights:Sender [Object]
```

**Comment by DonJayamanne** [comment]
Is it possible to provide to provide the sample code (share the repo, or privately share it with me). 
I will need the source to replicate this issue.

**Comment by egabrum** [comment]
@DonJayamanne To simulate a large test repository, you can just define a test with more than 5000 parameters:

```
#content of test_multi.py
import pytest

@pytest.yield_fixture(scope="module")
def moduleScoped_mFix():
   print "\nThis should only run once per module."
   
   
@pytest.yield_fixture(scope="class")
def classScoped_cFix():
   print "\nThis should only run once per class."
   
@pytest.yield_fixture(scope="function")
def classScoped_fFix():
   print "\nThis should only run once per function."

 
class TestClass_one():
   @pytest.mark.parametrize('y', range(5000))
   def test_manyParams(self, classScoped_cFix,y):
      assert y == y
```

In my case, it does not collect more that 4800 tests.

**Comment by DonJayamanne** [comment]
From the status bar click ```Run Tests``` and select the menu option ```Discover Unit Tests```.  
Next Ppease could you paste the contents of the python test panel...
You should see something like the following:
```
============================= test session starts ==============================
platform darwin -- Python 2.7.11, pytest-2.8.5, py-1.4.31, pluggy-0.3.1
rootdir: /Users/donjayamanne/Projects/Python/Learn, inifile: 
collected 5004 items
<Module 'test/test_cool_module.py'>
<Module 'test/test_three.py'>
  <Class 'TestClass_one'>
    <Instance '()'>
      <Function 'test_manyParams[0]'>
      <Function 'test_manyParams[1]'>
      <Function 'test_manyParams[2]'>
      <Function 'test_manyParams[3]'>
      <Function 'test_manyParams[4]'>





      <Function 'test_manyParams[4996]'>
      <Function 'test_manyParams[4997]'>
      <Function 'test_manyParams[4998]'>
      <Function 'test_manyParams[4999]'>

========================= no tests ran in 0.50 seconds =========================


```

**Comment by egabrum** [comment]
```
============================= test session starts =============================
platform win32 -- Python 2.7.13, pytest-3.0.7, py-1.4.33, pluggy-0.4.0 -- c:\python27\python.exe
cachedir: .cache
metadata: {'Python': '2.7.13', 'Platform': 'Windows-7-6.1.7601-SP1', 'Packages': {'py': '1.4.33', 'pytest': '3.0.7', 'pluggy': '0.4.0'}, 'Plugins': {'html': '1.14.2', 'timeout': '1.2.0', 'metadata': '1.3.0'}}
rootdir: c:\Users\sesa276778\OneDrive\SE-Casa\Python_local\sandbox\pytest, inifile:
plugins: timeout-1.2.0, metadata-1.3.0, html-1.14.2
collecting ... collected 5016 items
<Module 'fixtures/test_fixtureScopes.py'>
  <Class 'TestClass_one'>
    <Instance '()'>
      <Function 'test_one1'>
      <Function 'test_one2'>
  <Class 'TestClass_two'>
    <Instance '()'>
      <Function 'test_two1'>
      <Function 'test_two2[param1-y1-x1]'>
      <Function 'test_two2[param1-y1-x2]'>
      <Function 'test_two2[param1-y2-x1]'>
      <Function 'test_two2[param1-y2-x2]'>
      <Function 'test_two2[param2-y1-x1]'>
      <Function 'test_two2[param2-y1-x2]'>
      <Function 'test_two2[param2-y2-x1]'>
      <Function 'test_two2[param2-y2-x2]'>
      <Function 'test_manyParams[0]'>
      <Function 'test_manyParams[1]'>
      <Function 'test_manyParams[2]'>
      <Function 'test_manyParams[3]'>
      <Function 'test_manyParams[4]'>
<Module 'fixtures/test_multi.py'>
  <Class 'TestClass_one'>
    <Instance '()'>
      <Function 'test_manyParams2[0]'>
      <Function 'test_manyParams2[1]'>
      <Function 'test_manyParams2[2]'>
      <Function 'test_manyParams2[3]'>
```
[etc]
```
      <Function 'test_manyParams2[4750]'>
      <Function 'test_manyParams2[4751]'>
      <Function 'test_manyParams2[4752]'>
      <Function 'test_manyParams2[4753]'>
```
and ends there.

**Comment by egabrum** [comment]
Last release was more than a month ago. Any plans to release soon?
Just curious.

**Comment by egabrum** [comment]
Seems to be fixed in 0.6.5
thanks!

**Comment by DonJayamanne** [comment]
Closing as this has been confirmed to be fixed
